### PR TITLE
Fix `TimestampLeaseRequests` serialization

### PR DESCRIPTION
**Pull request by fsamuel-bs**
## General
**Before this PR**:
In integration tests we noticed that `TimestampLeaseRequests` would fail to de-serialize in Timelock with
```
com.palantir.conjure.java.undertow.runtime.FrameworkException: Failed to deserialize request
...
Cannot find a (Map) Key deserializer for type [simple type, class com.palantir.atlasdb.common.api.timelock.TimestampLeaseName]
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 1]
```

**After this PR**:

==COMMIT_MSG==
==COMMIT_MSG==

**Priority**:

**Concerns / possible downsides (what feedback would you like?)**:

**Is documentation needed?**:

## Compatibility
**Does this PR create any API breaks (e.g. at the Java or HTTP layers) - if so, do we have compatibility?**:

**Does this PR change the persisted format of any data - if so, do we have forward and backward compatibility?**:

**The code in this PR may be part of a blue-green deploy. Can upgrades from previous versions safely coexist? (Consider restarts of blue or green nodes.)**:

**Does this PR rely on statements being true about other products at a deployment - if so, do we have correct product dependencies on these products (or other ways of verifying that these statements are true)?**:

**Does this PR need a schema migration?**

## Testing and Correctness
**What, if any, assumptions are made about the current state of the world? If they change over time, how will we find out?**:

**What was existing testing like? What have you done to improve it?**:

**If this PR contains complex concurrent or asynchronous code, is it correct? The onus is on the PR writer to demonstrate this.**:

**If this PR involves acquiring locks or other shared resources, how do we ensure that these are always released?**:

## Execution
**How would I tell this PR works in production? (Metrics, logs, etc.)**:

**Has the safety of all log arguments been decided correctly?**:

**Will this change significantly affect our spending on metrics or logs?**:

**How would I tell that this PR does not work in production? (monitors, etc.)**:

**If this PR does not work as expected, how do I fix that state? Would rollback be straightforward?**:

**If the above plan is more complex than “recall and rollback”, please tag the support PoC here (if it is the end of the week, tag both the current and next PoC)**:

## Scale
**Would this PR be expected to pose a risk at scale? Think of the shopping product at our largest stack.**:

**Would this PR be expected to perform a large number of database calls, and/or expensive database calls (e.g., row range scans, concurrent CAS)?**:

**Would this PR ever, with time and scale, become the wrong thing to do - and if so, how would we know that we need to do something differently?**:

## Development Process
**Where should we start reviewing?**:

**If this PR is in excess of 500 lines excluding versions lock-files, why does it not make sense to split it?**:

**Please tag any other people who should be aware of this PR**:
@jeremyk-91
@raiju

### CODE REVIEW

File: timelock-api/src/test/java/com/palantir/atlasdb/timelock/api/TimestampLeaseRequestsTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import org.junit.jupiter.api.Test;
+
+public class TimestampLeaseRequestsTest {
+    private static final ObjectMapper SERIALIZATION_MAPPER =
+            ObjectMappers.newClientObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
+    private static final ObjectMapper DESERIALIZATION_MAPPER =
+            ObjectMappers.newServerObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
+
+    @Test
+    void canSerializeAndDeserialize() throws IOException {
```
Comment (jkozlowski) [inline timelock-api/src/test/java/com/palantir/atlasdb/timelock/api/TimestampLeaseRequestsTest.java:38]: It almost feels like instead we should get the proper integration test going, with Timelock wired up... But I understand we want to start testing without blocking on that

File: atlasdb-commons-api/src/main/java/com/palantir/atlasdb/common/api/timelock/TimestampLeaseName.java [hunk 2]
```diff
@@ -16,17 +16,14 @@
...
 import com.fasterxml.jackson.annotation.JsonValue;
-import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
-import com.fasterxml.jackson.databind.annotation.JsonSerialize;
 import com.palantir.logsafe.Preconditions;
 import com.palantir.logsafe.Safe;
 import com.palantir.logsafe.SafeArg;
 import java.util.Comparator;
 import org.immutables.value.Value;
 
-@JsonDeserialize(as = ImmutableTimestampLeaseName.class)
```
Comment (jkozlowski) [inline atlasdb-commons-api/src/main/java/com/palantir/atlasdb/common/api/timelock/TimestampLeaseName.java:28]: womp

File: timelock-api/build.gradle [hunk 3]
```diff
@@ -22,6 +23,7 @@ dependencies {
 subprojects {
     apply from: "../../gradle/shared.gradle"
     dependencies {
+        api project(':atlasdb-commons-api')
```
Comment (jkozlowski) [inline timelock-api/build.gradle:26]: ```tasks.checkImplicitDependenciesMain.enabled = false```

womp

Comment (jkozlowski) [reply timelock-api/build.gradle:26]: shall we FLUP with enabling this? I wonder how hard could this be...

**Review by jkozlowski — APPROVED** [review]
approving for RC

**Comment by fsamuel-bs** [comment]
Tested internally and this behaves as expected.

**Review by mdaudali — COMMENTED** [review]
Going through for second passes: Thanks!

**Pull request merged**